internal/core/config/config.go

```diff
@@ -19,8 +19,8 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-// Config is the top level config struct that everything goes under
-type Config struct {
+// CommonConfig is the top level config struct for configurations that are common to all platoforms
+type CommonConfig struct {
```

Comment:
This is backwards.  The main Config struct should stay the same and have the server fields embedded.  Or, even better, just use dynamic defaults provided by the [defaults library it is using](https://github.com/creasty/defaults) to set this at runtime based on whether it's running on Linux or Windows.